### PR TITLE
ci(Workflows): New versioning workflow [skip hint]

### PR DESCRIPTION
**Pull request by cbcrouse**
This commit will not affect the version of the published code. See [Commit Hints](https://github.com/cbcrouse/Versioning.NET/wiki/Commit-Hints).

**Many thanks for submitting your Pull Request :heart:!**

**What this PR does / why we need it**:
* Adds a new workflow which triggers on changes from the `main` branch that will result in the project version being incremented.

**Special notes for reviewers**:
* The URL for the repository might not be entirely correct, please review the tokens and username values. I tried to grab values from the [publish workflow](https://github.com/serverlessworkflow/synapse/blob/main/.github/workflows/publish.yml).

**Additional information (if needed):**

Due to the existing workflow that triggers on release, I did not add any additional workflows. If you decide to create a workflow in the future that triggers on the completion of the versioning, here is an example:

```yaml
name: main

on:

  workflow_run:
    workflows: [auto-version]
    types:
      - completed
    
  workflow_dispatch:
    
jobs:
    
  on-success:
    runs-on: ubuntu-latest
    if: ${{ github.event.workflow_run.conclusion == 'success' }}
    steps:
      - run: echo 'The triggering workflow passed'
      
  on-failure:
    runs-on: ubuntu-latest
    if: ${{ github.event.workflow_run.conclusion == 'failure' }}
    steps:
      - run: echo 'The triggering workflow failed'
```

---

I hope this helps, cheers!

### CODE REVIEW

**Comment by cdavernas** [comment]
Thank you very much for your contribution @cbcrouse ! And a big up for your awesome [Versioning.NET project](https://github.com/cbcrouse/Versioning.NET)!!!

**Review by cdavernas — APPROVED** [review]

File: .github/workflows/auto-version.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.4.2
+        with:
+          fetch-depth: 0
+
+      - name: Reset Origin
+        run: |
+          git remote set-url origin "https://cdavernas:${{ secrets.GITHUB_TOKEN }}@github.com/serverlessworkflow/synapse.git"
```
Comment (JBBianchi) [inline .github/workflows/auto-version.yml:20]: Questionning `cdavernas` in here, isn't there any alternative?

**Review by JBBianchi — APPROVED** [review]
LGTM besides the little comment

**Comment by cdavernas** [comment]
Hey @cbcrouse! Sorry for the delay for merging the your (awesome) PR. We needed to make sure to prepare our solution and purge past comments in order to start using proper versioning.
I'll try to push it in for next week ;)

File: .github/workflows/auto-version.yml (hunk 1 above)
Comment (cdavernas) [reply .github/workflows/auto-version.yml:20]: Not that I know of. it's, however, a very little constraint IMHO

Comment (0nary) [reply .github/workflows/auto-version.yml:20]: You can use `${{ github.actor }}` instead of `cdavernas` or `${{ github.triggering_actor }}`.

**Pull request merged**